Makefile

```diff
@@ -399,9 +402,28 @@ ifeq ($(BUILD_TYPE),metal)
 	cp go-llama/build/bin/ggml-metal.metal backend-assets/grpc/
 endif
 
-backend/cpp/llama/grpc-server:
-	LLAMA_VERSION=$(CPPLLAMA_VERSION) $(MAKE) -C backend/cpp/llama grpc-server
-
+## BACKEND CPP LLAMA 
+# Sets the variables in case it has to build the gRPC locally.
+INSTALLED_PACKAGES=$(CURDIR)/backend/cpp/grpc/installed_packages
+INSTALLED_LIB_CMAKE=$(INSTALLED_PACKAGES)/lib/cmake
+ADDED_CMAKE_ARGS=-Dabsl_DIR=${INSTALLED_LIB_CMAKE}/absl \
+                 -DProtobuf_DIR=${INSTALLED_LIB_CMAKE}/protobuf \
+                 -Dutf8_range_DIR=${INSTALLED_LIB_CMAKE}/utf8_range \
+                 -DgRPC_DIR=${INSTALLED_LIB_CMAKE}/grpc \
+                 -DCMAKE_CXX_STANDARD_INCLUDE_DIRECTORIES=${INSTALLED_PACKAGES}/include
+
+backend/cpp/llama/grpc-server:			
+ifdef BUILD_GRPC
+	backend/cpp/grpc/script/build_grpc.sh ${INSTALLED_PACKAGES}
```

Comment:
I'm wondering - here probably the best approach would be to use cmake directly.

I was reading initially https://github.com/grpc/grpc/tree/master/src/cpp#fetchcontent but leaving this for a follow-up, maybe you'd like to take a look first?

Reply:
Yes, I think that at least the bash-based solution allows people to build without docker, when the PR is approved.
Then, I can try to modify the gRPC installation using the CMake. 
Does it sound good?

Reply:
Sounds good!